src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -380,9 +381,49 @@ private static long ParseTimeValue(JToken jToken, string claimName)
         /// <param name="validationParameters">A <see cref="TokenValidationParameters"/>  required for validation.</param>
         /// <returns>Returns a <see cref="SecurityKey"/> to use for signature validation.</returns>
         /// <remarks>If key fails to resolve, then null is returned</remarks>
-        internal static SecurityKey ResolveTokenSigningKey(string kid, string x5t, TokenValidationParameters validationParameters)
+        private static SecurityKey ResolveTokenSigningKeyWithConfig(string kid, string x5t, TokenValidationParameters validationParameters)
         {
+            if (!string.IsNullOrEmpty(kid))
+            {             
+                if (validationParameters.Configuration?.SigningKeys != null)
```

Comment:
We don't need the null check here as this is called only when Configuration is non null.

Reply:
@brentschmaltz , I think it is good to perform null check in case new code is added in the future that does not perform null check first.

Reply:
I'm somewhat torn on what to do here. In a perfect world this method would only be called when the Configuration is non-null, but I'm not sure if we can expect the same in practice.

Do we want to throw a null reference exception here if it's null since this method should not be called without a Configuration present? I think having the null check could potentially be hiding an issue.

Reply:
I think throwing a NRE makes sense so it is consistent with the logic in many other methods.

Reply:
@mafurman, are you going to throw an exception?

Reply:
I'm ok with checking for null.